### PR TITLE
Simplify and clean build process

### PR DESCRIPTION
**Pull request by Aareksio**
## Description
The build process haven't been refreshed since the initial development 2 years ago. The changes aim to make the configuration simpler and replace old dependencies with latest recommendations.

The Pull Request is opened in WIP state to track CI reports and make sure no degradation occurred.

## Additional information
Please refer to commit messages for details of described changes.

## Checklist

- [x] My pull request is not a duplicate
- [x] I added a descriptive title to this pull request
- [x] My code follows the code style of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,16 +1,12 @@
 {
-	"name": "ASF-ui",
+	"name": "asf-ui",
 	"version": "0.0.0",
 	"description": "Standalone web interface for ASF",
 	"scripts": {
-		"dev": "webpack --config webpack.config.js --env development --progress",
-		"pre-build": "node scripts/generate-flags",
```
Comment (JustArchi) [inline package.json:7]: You no longer need `pre-build` for generating flags for localization?

**Comment by Aareksio** [comment]
@JustArchi @MrBurrBurr 
Could you please confirm that:

- Nothing depends on `name` field from package.json to be `ASF-ui`
- Scripts `dev`, `pre-build`, `deploy` and `generate-flags` were indeed not used
- All CIs use command `npm run(-script) build:dev`

**Comment by JustArchi** [comment]
> Nothing depends on name field from package.json to be ASF-ui

Nothing from me.

> Scripts dev, pre-build, deploy and generate-flags were indeed not used

I don't know how exactly `pre-build` was hooked into the process, but I thought it was used for generating flags for localization?

image

> All CIs use command npm run(-script) build:dev

All CIs use `build:ci`.

File: package.json (hunk 1 above)
Comment (Aareksio) [reply package.json:7]: From what I saw in `webpack.config.js` the script was run by `before-build-webpack` before build.

Comment (Aareksio) [reply package.json:7]: Confirmed, it generates `generated/flags.js` correctly. CIs seem to work just fine too.

**Comment by MrBurrBurr** [comment]
> Nothing depends on name field from package.json to be ASF-ui

Not that I know of. Its invalid node package name anyway since it contains capital letters.

**Pull request merged**